### PR TITLE
SystemXHR fix

### PR DESCRIPTION
**Pull request by mancas**
r? @mcjimenez

### CODE REVIEW

File: polyfills/systemxhr/systemxhr.js [hunk 1]
```diff
@@ -192,21 +193,61 @@
...
+            '', '');
+          value = document.implementation.createDocument('', 'html', doctype);
+          value.documentElement.innerHTML = this._response;
+          break;
+        case 'json':
+          value = JSON.parse(this._response);
+          break;
+        case 'blob':
+        case 'moz-blob':
+          value = new Blob(this._response);
```
Comment (mancas) [inline polyfills/systemxhr/systemxhr.js]: How can we retrieve the type of the blob from the response (string)?

File: polyfills/systemxhr/systemxhr.js [hunk 2]
```diff
@@ -192,21 +193,61 @@
...
         get: function() {
+          if (property === 'response') {
+            self._convertToResponseType();
+          }
           return this['_' + property];
         },
         set: function(value) {
           this['_' + property] = value;
-          navConnHelper.methodCall({
+          if (property !== 'responseType') {
```
Comment (mancas) [inline polyfills/systemxhr/systemxhr.js:203]: Let's assume that every request is made in plain text and then the client will convert the response to the appropriate type

File: polyfills/systemxhr/systemxhr.js [hunk 3]
```diff
@@ -192,21 +193,61 @@
...
                                     field: 'xhrId'
                                   }, property, value);
+          }
         }
       });
     });
 
+    this._convertToResponseType = function() {
+      if (typeof this._response !== 'string') {
+        return;
```
Comment (mcjimenez) [inline polyfills/systemxhr/systemxhr.js]: return this._response?

Comment (mcjimenez) [inline polyfills/systemxhr/systemxhr.js]: Also if what we're getting always is the responseText, why don't we store it as this._responseText?

File: services/systemxhr/js/app.js [hunk 4]
```diff
@@ -26,6 +26,10 @@
     // Params for the local operation:
     var opData = request.remoteData.data.params || [];
     var xhrId = request.remoteData.data.xhrId;
+    if (operation === 'open' && opData[2] === null ||
+      typeof opData[2] === 'undefined') {
```
Comment (mcjimenez) [inline services/systemxhr/js/app.js]: Nit: Indentation

File: tests/systemxhr.js [hunk 5]
```diff
@@ -40,7 +45,38 @@
...
           }
         };
+      }
 
+      function testXMLHttpRequestJSON() {
+        var newXHR = createXMLHttpRequest();
+
+        newXHR.onreadystatechange = evt => console.info(evt);
+        newXHR.open('get',
+                    'http://antonioma.github.io/WebAPI_pf/tests/test_list.json',
```
Comment (mcjimenez) [inline tests/systemxhr.js:55]: We should probably get something that's always on other origin (like www.google.com for example)

**Pull request merged**